### ISSUE TITLE
The input processing used to perturb the image, isn't it gradient regards to Entropy loss?

### ISSUE THREAD

**Issue by engkarat** [YU1ut/Ensemble-of-Leave-out-Classifiers#2]
Hi,
It seems like the original paper mentioned that the input processing is done regards to the entropy of the prediction. Is there any specific reason why do you use input processing regards to the cross entropy loss instead of the proposed one?
Thank you,
Engkarat

**Comment by YU1ut** [comment]
I just used the implementation of ODIN.
IMHO, decreasing the entropy of the prediction is the same as maximizing the largest element of the prediction (which is the cross entropy loss in ODIN). They both force the prediction to be more one-hot.
So I think there is no large difference between them.

**Comment by engkarat** [comment]
Thank you very much for your prompt reply. I totally agree with you. The reason I ask this question because I am also trying to reimplement this paper by myself. But, it is not working well. I am trying to find the difference.

Another point that mine is differently implemented is that I don't save the best model based on the validation AUROC between training. Is this really sensitive to the choice of the best model? Have you ever observe this before?

**Comment by YU1ut** [comment]
I have not tried it before. I am running my code again to see if this will happen.

**Comment by engkarat** [comment]
Thank you very much for your help.
In my case, when I train the model until the end at 100th epoch and use that as selected model, the AUROC is not as good as I expected. But when I follow your strategy (same as original paper) to save the best model base on the validation OOD dataset, it produces better performance, but not as good as reported. Now I am checking the difference again. If you have any update about your experiments, could you please share that to me. Thank you.

**Comment by YU1ut** [comment]
I also tried to use the latest model (100th epoch)and found the performance will drop a little but still acceptable.

Best AUROC Model:
FPR at TPR 95%:              1.93
Detection error:             3.07
AUROC:                      99.54
AUPR In:                    99.57
AUPR Out:                   99.52

Latest model:
FPR at TPR 95%:              2.59
Detection error:             3.55
AUROC:                      99.41
AUPR In:                    99.43
AUPR Out:                   99.42

**Comment by engkarat** [comment]
Thank you for sharing me this information.

**Comment by engkarat** [comment]
Let me further ask a question.

How about the dropped performance in case of CIFAR-100? Do they change more significant if 100th epoch is chosen? I supposed the different might be larger especially in WRN-28-10 rather than Dense-BC. Have you ever observe this case?

**Comment by YU1ut** [comment]
OK. I am trying it.

**Comment by YU1ut** [comment]
Yes. The change is more significant on CIFAR-100.

**Comment by engkarat** [comment]
Thank you very much for your prompt reply and help.

**Comment by apoorv2904** [comment]
Hi,
I am one of the authors of this paper.  Thank you for providing the implementation for the paper. Unfortunately, due to legal reasons, I cannot share the code. As pointed out, for CIFAR-100. the results are sensitive to which model is used.  Also, just using the model with the best AUROC will hurt the classification performance on in-distribution. 

In order to get the best of both worlds, we were saving the model with the highest AUROC within about 2% classification accuracy on the in-distribution validation set. This ensures that we don't suffer on classification performance and get good performance on OOD detection. I hope this helps
